### PR TITLE
Move users of `factory.NewConfigFactory` to `scheduler.New`

### LINKED ISSUE #71859: [scheduler cleanup phase 2]: Move users of `factory.NewConfigFactory` to `scheduler.New`
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.

Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

If you contributed in phase 1, please give others a chance before claiming this issue. You can also help out by reviewing the phase 2 PRs.

Our existing, multi-step constructor `NewConfigFactory` is being replaced by the simpler `New` function. (See https://github.com/kubernetes/kubernetes/pull/69057) This issue covers migrating all users to the new function.

Today, all callers appear to be in either `cmd/kube-scheduler` or `test/integration`.

When the migration is complete, we should also be able to,

- remove `NewSchedulerConfig` from `cmd/kube-scheduler/app`
- remove `scheduler.NewFromConfigurator`
- rename `scheduler.NewFromConfig` to `scheduler.newFromConfig`

/sig scheduling
/kind cleanup
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by wgliang**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
 /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #71859

**Special notes for your reviewer**:
/sig scheduling

**Does this PR introduce a user-facing change?**:

```release-note
Move users of `factory.NewConfigFactory` to `scheduler.New`.
```

### CODE REVIEW

**Comment by wgliang** [comment]
@misterikkit 
note：I am afraid that `scheduler.NewFromConfig` needs to be kept public, and there are still scenarios for external use, such as integration testing.

**Comment by wgliang** [comment]
@misterikkit There are some integration like `TestSchedulerCreationFromConfigMap` tests for create scheduler form config. If we private `NewFromConfig` that these code maybe useless.

**Comment by misterikkit** [comment]
Hmm, I see your point. Can we merge this PR but keep the issue open? I'd like us to explore what it would take to get rid of 'NewFromConfig'

File: test/integration/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -36,7 +36,6 @@ import (
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
-	schedulerapp "k8s.io/kubernetes/cmd/kube-scheduler/app"
 	schedulerappconfig "k8s.io/kubernetes/cmd/kube-scheduler/app/config"
```
[the remark this answers is not in the document]
Comment (wgliang) [reply test/integration/scheduler/scheduler_test.go]: Done. You are right, thanks. :)

**Review by misterikkit — COMMENTED** [review]
Could you fix the git commit message? Otherwise,
/lgtm

**Comment by wgliang** [comment]
@misterikkit 
Sorry,It is my fault. I have fixed it.

**Review by bsalamat — COMMENTED** [review]
/approve

**Comment by misterikkit** [comment]
@kubernetes/sig-testing-pr-reviews need someone to review & approve of integration test changes.

File: test/integration/util/util.go [hunk 2]
```diff
@@ -69,12 +68,8 @@ func StartScheduler(clientSet clientset.Interface) (factory.Configurator, Shutdo
 	stopCh := make(chan struct{})
 	schedulerConfigurator := createSchedulerConfigurator(clientSet, informerFactory, stopCh)
 
-	sched, err := scheduler.NewFromConfigurator(schedulerConfigurator, func(conf *factory.Config) {
-		conf.Recorder = evtBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "scheduler"})
-	})
-	if err != nil {
-		klog.Fatalf("Error creating scheduler: %v", err)
-	}
+	config, _ := schedulerConfigurator.CreateFromConfig(schedulerapi.Policy{})
```
Comment (ixdy) [inline test/integration/util/util.go]: should we be discarding the error here?

Comment (wgliang) [reply test/integration/util/util.go]: No, we need to check this error, I have changed it. 
Thanks for your reminder and review.

**Review by misterikkit — APPROVED** [review]
/lgtm

**Comment by wgliang** [comment]
@kubernetes/sig-testing-pr-reviews need someone to review & approve of integration test changes.

File: test/integration/scheduler/scheduler_test.go [hunk 3]
```diff
@@ -242,29 +245,33 @@ func TestSchedulerCreationFromNonExistentConfigMap(t *testing.T) {
...
-		Recorder:        eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: v1.DefaultSchedulerName}),
-		Broadcaster:     eventBroadcaster,
-	}
+		nil,
+		scheduler.WithName(v1.DefaultSchedulerName),
+		scheduler.WithHardPodAffinitySymmetricWeight(v1.DefaultHardPodAffinitySymmetricWeight),
+		scheduler.WithBindTimeoutSeconds(defaultBindTimeout))
 
-	_, err := schedulerapp.NewSchedulerConfig(ss.Complete())
+	// _, err := schedulerapp.NewSchedulerConfig(ss.Complete())
```
Comment (misterikkit) [inline test/integration/scheduler/scheduler_test.go]: I just noticed this line. Should it be deleted?

**Review by misterikkit — CHANGES_REQUESTED** [review]

**Comment by ixdy** [comment]
/approve

but I agree with @misterikkit's comment

File: test/integration/scheduler/scheduler_test.go (hunk 3 above)
Comment (wgliang) [reply test/integration/scheduler/scheduler_test.go]: Done. thanks.

**Comment by wgliang** [comment]
@ixdy Thanks.

File: test/integration/scheduler/scheduler_test.go [hunk 4]
```diff
@@ -179,33 +177,38 @@ func TestSchedulerCreationFromConfigMap(t *testing.T) {
...
-				HardPodAffinitySymmetricWeight: v1.DefaultHardPodAffinitySymmetricWeight,
-				SchedulerName:                  v1.DefaultSchedulerName,
-				AlgorithmSource: kubeschedulerconfig.SchedulerAlgorithmSource{
-					Policy: &kubeschedulerconfig.SchedulerPolicySource{
-						ConfigMap: &kubeschedulerconfig.SchedulerPolicyConfigMapSource{
-							Namespace: policyConfigMap.Namespace,
-							Name:      policyConfigMap.Name,
-						},
+
+		sched, err := scheduler.New(clientSet,
```
Comment (krmayankk) [inline test/integration/scheduler/scheduler_test.go]: we need this call in extender_test.go as well

**Review by misterikkit — COMMENTED** [review]
/lgtm

**Comment by misterikkit** [comment]
:heart: :champagne: :smiley:

**Comment by bsalamat** [comment]
Awesome! Thanks, @wgliang and @misterikkit! :tada:

File: test/integration/scheduler/scheduler_test.go (hunk 4 above)
Comment (krmayankk) [reply test/integration/scheduler/scheduler_test.go]: extender_test.go uses initTestScheduler, which in turn finally calls scheduler.NewFromConfig() in test/integration/scheduler/util.go

Comment (krmayankk) [reply test/integration/scheduler/scheduler_test.go]: start here https://github.com/kubernetes/kubernetes/blob/master/test/integration/scheduler/extender_test.go#L352

Comment (misterikkit) [reply test/integration/scheduler/scheduler_test.go]: Ah, that's a good point. Looks like we'll need a follow-up PR to get `test/integration/util` onto the new constructor.

**Comment by misterikkit** [comment]
I suspect this PR broke the following test.
```sh
make test-integration WHAT="./test/integration/scheduler_perf" KUBE_TEST_ARGS="-run xx -bench Node/500/250 "
```

**Pull request merged**